ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/WrappingRuleTest.kt

```diff
@@ -1871,6 +1871,303 @@ internal class WrappingRuleTest {
+            import java.util.HashMap
+            """.trimIndent()
+        wrappingRuleAssertThat(code)
+            .hasLintViolation(1, 28, "Missing newline after \";\"")
+            .isFormattedAs(formattedCode)
+    }
+
+    @Test
+    fun `Issue 1078 - Given a multiline semi separated with multiple items`() {
+        val code =
+            """
+            fun test() {
+                val a = 0; val b = 0; fun bar() {
+                    // no-op
+                }; for(i in 0..10) {
+                    println(i); println(i); a++; println(a)
+                }
+            }
+            """.trimIndent()
+        val formattedCode =
+            """
+            fun test() {
+                val a = 0;
+                val b = 0;
+                fun bar() {
+                    // no-op
+                };
+                for(i in 0..10) {
+                    println(i);
+                    println(i);
+                    a++;
+                    println(a)
+                }
+            }
+            """.trimIndent()
+        wrappingRuleAssertThat(code)
+            .hasLintViolations(
+                LintViolation(2, 15, "Missing newline after \";\""),
+                LintViolation(2, 26, "Missing newline after \";\""),
+                LintViolation(4, 7, "Missing newline after \";\""),
+                LintViolation(5, 20, "Missing newline after \";\""),
+                LintViolation(5, 32, "Missing newline after \";\""),
+                LintViolation(5, 37, "Missing newline after \";\""),
+            )
+            .isFormattedAs(formattedCode)
+    }
+
+    @Test
+    fun `Issue 1078 - Given a enum with semicolon has not violation`() {
+        val code =
+            """
+            enum class INDEX1 { ONE, TWO, THREE }
+            enum class INDEX2 { ONE, TWO, THREE; }
+            enum class INDEX3 { ONE, TWO, THREE; /* with comment */ }
+            enum class INDEX4 {
+                ONE, TWO, THREE;
+            }
+            enum class INDEX5 {
+                ONE,
+                TWO,
+                THREE;
+            }
+            enum class INDEX6 {
+                ONE,
+                TWO,
+                THREE,
+                ;
+            }
+            enum class INDEX7 {
+                ONE,
+                TWO,
+                THREE,
+                ;
+                fun foo() = ""
+            }
+            """.trimIndent()
+        wrappingRuleAssertThat(code)
+            .hasNoLintViolations()
+    }
+
+    @Test
+    fun `Issue 1078 - Given a companion object with semicolon has not violation`() {
+        val code =
+            """
+            class FirstClass(private val param: String) {
+                companion object;
+                private var toto: Boolean = false
+            }
+
+            class SecondClass(private val param: String) {
+                companion object; // with comment
+                private var toto: Boolean = false
+            }
```

Comment:
What is tested in this method that was not tested in the other tests?

Reply:
`companion object; // with comment` here `companion object;` is allowed I just added additional `// with comment` comment after the semi

Reply:
I have added a new TC with comments in the code